File: ts/src/342.ts
//2264. Largest 3-Same-Digit Number in String
function largestGoodInteger(num: string): string {
     let ans = "";

        for (let i = 0; i + 2 < num.length; i++) {
            if (num[i] === num[i + 1] && num[i] === num[i + 2]) {
                const triplet = num.substring(i, i + 3);
                if (triplet > ans) {
                    ans = triplet;
                }
            }
        }

        return ans;
};
//sub: https://leetcode.com/problems/largest-3-same-digit-number-in-string/submissions/1734435516/?envType=daily-question&envId=2025-08-14
//sol: https://leetcode.com/problems/largest-3-same-digit-number-in-string/solutions/7077957/2264-largest-3-same-digit-number-in-stri-r0lp/

File: rust/src/342.rs
//342. Power of Four
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        if n <= 0 {
            return false;
        }
        let x = (n as f64).ln() / 4f64.ln();
        let diff = (x - x.round()).abs();
        diff < 1e-10        
    }
}
//sub: https://leetcode.com/problems/power-of-four/submissions/1735523233/?envType=daily-question&envId=2025-08-15
//sol: https://leetcode.com/problems/power-of-four/solutions/7081991/342-power-of-four-by-nalindalal2004-1gfn/ 
